source/core/parameters.cc

```diff
@@ -2375,13 +2383,179 @@ namespace Parameters
                                          rbf_data["node_z"].begin(),
                                          rbf_data["node_z"].end());
                 }
+              particles[i].initialize_shape(shape_type, shape_arguments);
+            }
+          else if (shape_type == "composite")
+            {
+              if (shape_arguments_str == "1") // Default case
+                {
+                  // Default
+                  shape_type      = "sphere";
+                  shape_arguments = std::vector<double>(1.);
```

Comment:
Maybe it would be nice to add a warning that the filename of the composite is not provided and the default value is a sphere. No one will use a composite shape type to have a default sphere and not having the filename would be probably forgotten by the user.

Reply:
Agreed, I would just throw an exception here. For example you could have a default filename that is garbage (dumb_dumb_file, whatev) and if the filename is dumb_dumb_file you could throw an exception.